src/iop/denoiseprofile.c

```diff
@@ -40,6 +41,9 @@
 #include <xmmintrin.h>
 #endif
 
+// which version of the non-local means code should be used?  0=old (this file), 1=new (src/common/nlmeans_core.c)
+#define USE_NEW_IMPL 1
```

Comment:
If USE_NEW_IMPL is defined and #HAVE_OPENCL is defined, this creates an introspection error with a clang build, because all 6 calls to the sign() function have been removed from the code path.

If the NEW_IMPLementation is intended to work in a build without OpenCL, this needs a fix... if it's NOT intended to work without OpenCL, then it's probably good news for the build to fail, but maybe there's a way to make it fail gracefully and indicate the root issue?

I put some thought to a fix but I am not sure what is best to defend against the case of someone adding another feature later which calls sign() after all! Is there a pragma that can make the introspection failure into just a warning?

Reply:
> I didn't want to touch that because I don't have OpenCL even though I do compile with it included.

You will be pleased to hear that meanwhile I built this branch with clang and OpenCL on Windows (!) and gave this a good go with a dozen noisy photos (Pentax, motorsport, and heavy rain are not a good combination for noise :-) ) with no problems.